components/clarinet-format/src/formatter/mod.rs

```diff
@@ -0,0 +1,1232 @@
+// options always on new lines
+// Functions Always on multiple lines, even if short
+fn format_function(settings: &Settings, exprs: &[PreSymbolicExpression]) -> String {
+    let func_type = display_pse(settings, exprs.first().unwrap(), "");
+    let indentation = &settings.indentation.to_string();
+    let args_indent = format!("{}{}", indentation, indentation);
+
+    let mut acc = format!("({func_type} (");
+
+    // function name and arguments
+    if let Some(def) = exprs.get(1).and_then(|f| f.match_list()) {
+        if let Some((name, args)) = def.split_first() {
+            acc.push_str(&display_pse(settings, name, ""));
+
+            let mut iter = args.iter().peekable();
+            while let Some(arg) = iter.next() {
+                let trailing = get_trailing_comment(arg, &mut iter);
+                if arg.match_list().is_some() {
+                    // expr args
+                    acc.push_str(&format!(
+                        "\n{}{}",
+                        args_indent,
+                        format_source_exprs(settings, &[arg.clone()], &args_indent)
+                    ))
+                } else {
+                    // atom args
+                    acc.push_str(&format_source_exprs(settings, &[arg.clone()], &args_indent))
+                }
+                if let Some(comment) = trailing {
+                    acc.push(' ');
+                    acc.push_str(&display_pse(settings, comment, ""));
+                }
+            }
+            if args.is_empty() {
+                acc.push(')')
+            } else {
+                acc.push_str(&format!("\n{})", indentation))
+            }
+        } else {
+            panic!("can't have a nameless function")
+        }
+    }
+
+    // function body expressions
+    for expr in exprs.get(2..).unwrap_or_default() {
+        acc.push_str(&format!(
+            "\n{}{}",
+            indentation,
+            format_source_exprs(settings, &[expr.clone()], &settings.indentation.to_string(),)
+        ))
+    }
+    acc.push_str("\n)\n\n");
+    acc.to_owned()
+}
+
+// This code handles the line width wrapping and happens near the bottom of the
+// traversal
+fn to_inner_content(
+    list: &[PreSymbolicExpression],
+    settings: &Settings,
+    previous_indentation: &str,
+) -> String {
+    let mut result = String::new();
+    let mut current_line_width = previous_indentation.len();
+    let mut first_on_line = true;
+    let base_indent = format!(
+        "{}{}",
+        previous_indentation,
+        settings.indentation.to_string()
+    );
+
+    for expr in list.iter() {
+        let formatted = format_source_exprs(settings, &[expr.clone()], &base_indent);
+        let trimmed = t(&formatted);
+        let expr_width = trimmed.len();
+
+        if !first_on_line {
+            // For subexpressions that would exceed line length, add newline with increased indent
+            if current_line_width + expr_width + 1 > 80 {
```

Comment:
Consider changing the hardcoded value of `80` to use `settings.max_line_length` instead, so that formatting is configurable.